### ISSUE TITLE
Ability to using globbing for template directories

### ISSUE THREAD

**Issue by newtriks** [HenrikJoreteg/templatizer#74]
I like to organise my templates into the same directory as my modules. It would be useful I think to be able to introduce some globbing logic to include jade templates from multiple directories:

```
templatizer([
  path.join(appDir, '/**/*.jade')
], path.join(appDir, '/templates.js'));
```

Currently I get around this simply by providing an array but I have to update my config each time I add new modules, no biggie I know...

```
templatizer([
  path.join(appDir, '/app/templates'),
  path.join(appDir, '/home/templates'),
  path.join(appDir, '/info/templates')
], path.join(appDir, '/templates.js'));
```

-> [Example app](https://github.com/newtriks/ampersand-example/blob/master/moonbootsConfig.js#L27).

**Comment by lukekarrys** [comment]
Thanks for the issue @newtriks! I could see this being useful, especially in the scenario you described. I don't have the time right now to implement this, but I'd gladly accept a PR with this functionality. Any interest in trying this out and submitting a PR? :smile_cat:

**Comment by newtriks** [comment]
@lukekarrys nice one. Busy this end too but will gladly have a dive in on a quiet evening and see if I can contribute in any way :+1:

**Comment by newtriks** [comment]
@lukekarrys so looking through the source code it maybe perfectly feasible to completely replace the template directory string with a glob. Maybe using [node-glob](https://github.com/isaacs/node-glob) to perform a synchronous glob search we could replace both the[ array iteration](https://github.com/HenrikJoreteg/templatizer/blob/master/templatizer.js#L75) and  [walkdir logic](https://github.com/HenrikJoreteg/templatizer/blob/master/templatizer.js#L81). Supporting both a template directory string and/or a glob pattern I think would be overkill so would it be feasible to simply go for the glob as the first argument?

**Comment by lukekarrys** [comment]
@newtriks I think that should work. If you pass node-glob a path without any special characters, does it just return a single item array containing that path? If so, that'll work, but it still needs to support the original behavior where you just pass a directory path and it finds all `.jade` files in there. Hopefully, this can just be a minor version bump, I'd rather not have to change the existing API and do a major version bump.

**Comment by newtriks** [comment]
> If you pass node-glob a path without any special characters, does it just return a single item array containing that path?

@lukekarrys yup. Ok cool, I thought a breaking API change wouldn't be ideal :)

**Comment by lukekarrys** [comment]
Closed by #75`
